pkg/route/secretmanager/manager_test.go

```diff
@@ -0,0 +1,300 @@
+
+	"github.com/openshift/library-go/pkg/route/secret/fake"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type routeSecret struct {
+	routeName  string
+	secretName string
+}
+
+func TestRegisterRoute(t *testing.T) {
+	namespace := "ns"
+
+	scenarios := []struct {
+		name               string
+		rs                 []routeSecret
+		sm                 fake.SecretMonitor
+		expectHandlersKeys []string
+		expectErr          int
+	}{
+		{
+			name: "route can be registered only once with any secret",
+			rs: []routeSecret{
+				{routeName: "route", secretName: "secret"},
+			},
+			expectHandlersKeys: []string{namespace + "/route"},
+			expectErr:          0,
+		},
+		{
+			name: "same route cannot be registered again with same secret",
+			rs: []routeSecret{
+				{routeName: "route1", secretName: "secret1"},
+				{routeName: "route1", secretName: "secret1"},
+			},
+			expectHandlersKeys: []string{namespace + "/route1"},
+			expectErr:          1,
+		},
+		{
+			name: "same route cannot be registered again with different secrets",
+			rs: []routeSecret{
+				{routeName: "route1", secretName: "secret1"},
+				{routeName: "route1", secretName: "secret2"},
+				{routeName: "route1", secretName: "secret3"},
+			},
+			expectHandlersKeys: []string{namespace + "/route1"},
+			expectErr:          2,
+		},
+		{
+			name: "different routes can be registered with same secret",
+			rs: []routeSecret{
+				{routeName: "route1", secretName: "secret1"},
+				{routeName: "route2", secretName: "secret1"},
+			},
+			expectHandlersKeys: []string{namespace + "/route1", namespace + "/route2"},
+			expectErr:          0,
+		},
+		{
+			name: "different routes can be registered with different secrets",
+			rs: []routeSecret{
+				{routeName: "route1", secretName: "secret1"},
+				{routeName: "route2", secretName: "secret2"},
+			},
+			expectHandlersKeys: []string{namespace + "/route1", namespace + "/route2"},
+			expectErr:          0,
+		},
+		{
+			name: "error while adding SecretEventHandler",
+			rs: []routeSecret{
+				{routeName: "route", secretName: "secret"},
+			},
+			sm: fake.SecretMonitor{
+				Err: fmt.Errorf("some error"),
+			},
+			expectErr: 1,
+		},
+	}
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			mgr := NewManager(nil, nil).WithSecretMonitor(&s.sm)
+
+			gotErr := 0
+			for _, rs := range s.rs {
+				if err := mgr.RegisterRoute(context.TODO(), namespace, rs.routeName, rs.secretName); err != nil {
+					t.Log(err)
+					gotErr += 1
+				}
+			}
+			if gotErr != s.expectErr {
+				t.Fatalf("expected %d errors, got %d errors", s.expectErr, gotErr)
+			}
+			if len(s.expectHandlersKeys) != len(mgr.registeredHandlers) {
+				t.Fatalf("expected %d keys: %v, got %d keys: %v", len(s.expectHandlersKeys), s.expectHandlersKeys, len(mgr.registeredHandlers), mgr.registeredHandlers)
+			}
+			for _, key := range s.expectHandlersKeys {
+				if _, exists := mgr.registeredHandlers[key]; !exists {
```

Comment:
Why the value is not checked too?

Reply:
https://github.com/openshift/library-go/pull/1626#discussion_r1570797736